docs/proposals/images/machinepool-machines/inframachinepool-create-replicas.plantuml

```diff
@@ -0,0 +1,17 @@
+@startuml inframachinepool-create-replicas
+:InfraMachinePool create replicas reconciliation;
+if (status.InfrastructureMachineKind is set on InfraMachinePool) then (no)
+  :set status.InfrastructureMachineKind to InfraMachinePoolMachine;
+else (yes)
+endif
+repeat 
+  :create provider specific resource representing a replica;
+repeat while (~# replicas < # desired replicas ?) is (yes)
+->no;
+repeat 
+  :create InfraMachinePoolMachine representing a replica;
+repeat while (every replica has an InfraMachinePoolMachine?) is (no)
+->yes;
+:continue reconciling InfraMachinePool...;
```

Comment:
not sure what this represents, it may cause confusion

Reply:
That was just leftover from the original diagram [here](https://github.com/kubernetes-sigs/cluster-api/blob/5db6e22519699f4a563795ed2ac1514d4675346d/docs/proposals/images/machinepool-machines/infraref-changed-sequence.plantuml#L15). I think the intention was to have the controller do whatever it originally was supposed to do after the replica creation.

Reply:
For MachinePool Machines, though, this would probably be to delete the outdated/excess replicas, i.e. leading into the second diagram.